### ISSUE TITLE
No Output After Training on Weak Supervision

### ISSUE THREAD

**Issue by theahmedmustafa** [autonise/CRAFT-Remade#54]
I tried training for Weak supervision using the SynthText Strong supervision model on a custom data set.

I trained for just three epochs with 1000 iterations each but when I use the models that get saved at each iteration at _save-path_  the inference results are totally blank! No bounding box is drawn on any image after prediction.

Here are a few other observations:

1. The _Test-(iteration)_ folders in _save-path_ contains images from my test dateset with **blue** bounding boxes which I assume is the ground truth, but no predicted bounding box seem to be drawn
2. My loss drops to 0 and my accuracy goes to one just after a few iterations in the first epoch
3. My precision and recall are both 1.0. The cumulative F-score is **nan**, probably because at some stage it gives a division error for some reason.
4. All the folders in both _(iteration)-next-target_ and _(iteration)-predicted_  folders in the _Generated_ folder at _target-path_  are empty
5. Prediction on the SynthText Strong Supervision model works just fine

I did write a script to convert the annotations for my data to the ICDAR format (basically what the _pre_process_ function in _main.py_ does), and it seems to match the required format, so I do not think that is the issue. The training runs without error.

Also, all the text annotations for the bounding boxes in my ground truth are **###** since the data is not text level annotated, only the bounding boxes are drawn. I hope that is not the issue.

If you can make any sense of what the problem might be, kindly do tell me.
Regards

**Comment by mayank-git-hub** [comment]
Hello @theahmedmustafa .

Text annotations are required for weak supervision as it uses the number of characters in the word to break the word bounding box into equal sized segments. If no text annotations are provided, the model will just be fine-tuned to predict the output of the targets generated using the strong supervised model, which can have a lot of errors. This will cause instability during training and could lead to the results you are getting.

**Comment by theahmedmustafa** [comment]
I added the text annotations to my data set and it has started to produce bounding boxes, however I am confused why all the folders inside _Generated_ folder at _target-path_ are always empty
